### PR TITLE
[dotnet new] Make it possible to disable sdk installed templates

### PR DESCRIPTION
**Pull request by DavidKarlas**
aspnetcore team wants to install their locally templates when executing tests, this gives them ability to disable SDK installed templates, because we don't allow uninstalling SDK templates anymore

### CODE REVIEW

**Review by vlada-shubina — APPROVED** [review]

File: src/Cli/dotnet/commands/dotnet-new/NewCommandShim.cs
Comment (jonathanpeppers) [inline src/Cli/dotnet/commands/dotnet-new/NewCommandShim.cs:63]: `typeof(FileMetadataSearchSource).Assembly` was removed, was it intentional?

We are seeing `dotnet new android` started to fail here: https://github.com/xamarin/xamarin-android/pull/5849

`dotnet new` isn't finding our templates in `dotnet/template-packs`. Would this cause this issue? Thanks!

Comment (DavidKarlas) [reply src/Cli/dotnet/commands/dotnet-new/NewCommandShim.cs:63]: Sorry about breaking you, it was different change that broke you... It will be fixed via https://github.com/dotnet/sdk/pull/16967 just waiting for one more approval from @wli3 since he is more familiar with option workloads. Again sry.

Comment (jonathanpeppers) [reply src/Cli/dotnet/commands/dotnet-new/NewCommandShim.cs:63]: No problem, thanks.

**Pull request merged**
